From 1774ad6ef53312c636dd241956cee8ba9f39210b Mon Sep 17 00:00:00 2001
From: Evan Huus <eapache@gmail.com>
Date: Thu, 16 Jul 2015 22:34:14 -0400
Subject: [PATCH 03/12] wmem: don't remove tiny blocks from the recycler

They were never put in it in the first place because they can't hold the
necessary pointers.

Bug: 11373
Change-Id: I9e2ec76850929b5ac86e6f7a344d70f56ad3911c
Reviewed-on: https://code.wireshark.org/review/9673
Reviewed-by: Evan Huus <eapache@gmail.com>
(cherry picked from commit 1abc148210cb5c28b24770d1c5897babb43eb581)
Reviewed-on: https://code.wireshark.org/review/10583
---
 epan/wmem/wmem_allocator_block.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/epan/wmem/wmem_allocator_block.c b/epan/wmem/wmem_allocator_block.c
index e5be6b8..6d7d42a 100644
--- a/epan/wmem/wmem_allocator_block.c
+++ b/epan/wmem/wmem_allocator_block.c
@@ -591,7 +591,7 @@ wmem_block_split_free_chunk(wmem_block_allocator_t *allocator,
         if (chunk == allocator->master_head) {
             wmem_block_pop_master(allocator);
         }
-        else {
+        else if (WMEM_CHUNK_DATA_LEN(chunk) >= sizeof(wmem_block_free_t)) {
             wmem_block_remove_from_recycler(allocator, chunk);
         }
         return;
-- 
2.1.4

